sos/db/src/main/java/se/inera/intyg/common/db/v1/model/converter/certificate/QuestionOsakertDodsdatum.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2022 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.common.db.v1.model.converter.certificate;
+
+import static se.inera.intyg.common.sos_parent.support.RespConstants.DODSDATUM_JSON_ID;
+import static se.inera.intyg.common.sos_parent.support.RespConstants.DODSDATUM_OSAKERT_DELSVAR_ID;
+import static se.inera.intyg.common.sos_parent.support.RespConstants.DODSDATUM_SAKERT_DELSVAR_ID;
+import static se.inera.intyg.common.sos_parent.support.RespConstants.DODSDATUM_SAKERT_JSON_ID;
+import static se.inera.intyg.common.support.facade.util.ValidationExpressionToolkit.equalsWith;
+import static se.inera.intyg.common.support.facade.util.ValidationExpressionToolkit.singleExpression;
+import static se.inera.intyg.common.support.facade.util.ValueToolkit.uncertainDateValue;
+
+import java.util.List;
+import se.inera.intyg.common.services.texts.CertificateTextProvider;
+import se.inera.intyg.common.support.facade.model.Certificate;
+import se.inera.intyg.common.support.facade.model.CertificateDataElement;
+import se.inera.intyg.common.support.facade.model.config.CertificateDataConfigUncertainDate;
+import se.inera.intyg.common.support.facade.model.validation.CertificateDataValidation;
+import se.inera.intyg.common.support.facade.model.validation.CertificateDataValidationMandatory;
+import se.inera.intyg.common.support.facade.model.validation.CertificateDataValidationShow;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataUncertainDateValue;
+import se.inera.intyg.common.support.model.InternalDate;
+
+
+public class QuestionOsakertDodsdatum {
+
+    public static CertificateDataElement toCertificate(String value, int index, CertificateTextProvider texts) {
+        return CertificateDataElement.builder()
+            .id(DODSDATUM_OSAKERT_DELSVAR_ID)
+            .parent(DODSDATUM_SAKERT_DELSVAR_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigUncertainDate.builder()
+                    .id(DODSDATUM_JSON_ID)
+                    .allowedYears(List.of("2022", "2021"))
```

Comment:
SHould this be hardcoded?

Reply:
No, but was hardcoded initially for simplicity going forward but is now implemented as it should.